[AI-6534] Add Dell Powerflex Integration (#23183)
DataDog/integrations-core: dell_powerflex/README.md

> # Dell PowerFlex

Reviewer: The standard integration template uses the form `# Agent Check: {integration_name}`. Suggested:

```suggestion
# Agent Check: Dell PowerFlex
```

Also confirm the product name capitalization — Dell's official branding is "PowerFlex" (one word, mid-caps). Worth verifying that "Dell PowerFlex" is the intended display name throughout.

Author: Yes, Dell PowerFlex is the intended name throughout!

## Overview

> This check monitors [integration name] through the Datadog Agent. 

Reviewer: `[integration name]` is an unfilled template placeholder, and the link target is missing.

```suggestion
This check monitors [Dell PowerFlex][1] through the Datadog Agent.
```

Add `[1]: https://www.dell.com/en-us/dt/storage/powerflex/index.htm` (or the appropriate Dell product page) in the reference section at the bottom of the file.

> To learn more, visit https://docs.datadoghq.com/integrations/Dell PowerFlex/

Reviewer: Two issues with this line:

1. The URL contains a literal space character (`Dell PowerFlex`) — URLs can't contain spaces. If kept, the slug should be `dell_powerflex` (matching the integration directory).
2. This is a self-referential link. The README *is* the docs page at `/integrations/dell_powerflex/`, so a "visit our docs" link back to itself shouldn't be in the README.

Suggest removing the line entirely:

```suggestion
```

## Setup

> Visit our documentation to learn more. 

Reviewer: This sentence uses first person ("our") and is too vague — it doesn't tell the reader where to go or what to do.

This line should be replaced by the standard Setup content from the [integration README template](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/tooling/templates/integration/check/%7Bcheck_name%7D/README.md), with `### Installation`, `### Configuration`, and `### Validation` subsections.
